rest-api/dao/physical_measurements_dao.py

```diff
@@ -46,13 +48,18 @@ def get_with_session(self, session, obj_id, **kwargs):
       ParticipantDao().validate_participant_reference(session, result)
     return result
 
-  def get_with_children(self, physical_measurements_id):
+  def get_with_children(self, physical_measurements_id, for_update=False):
     with self.session() as session:
       query = session.query(PhysicalMeasurements) \
           .options(subqueryload(PhysicalMeasurements.measurements).subqueryload(
               Measurement.measurements)) \
           .options(subqueryload(PhysicalMeasurements.measurements).subqueryload(
               Measurement.qualifiers))
+
+      if for_update:
+        query = query.with_for_update()
+        return session, query.get(physical_measurements_id)
```

Comment:
You shouldn't be returning session from here. It doesn't match the return type below, and the session is useless once you exit here anyway. (The with self.session() as session will close the session when you exit the block.)

Instead you need a version of this that takes session as an argument... call with self.session() as session outside of the method and pass it in.